### PR TITLE
Adds landmines

### PR DESCRIPTION
**Pull request by Alberyk**
Adds landmines, objects that can be deployed and will trigger a certain effect when someone walks over them.

Can only be "disarmed" via destroying it, by bullets, explosion and etc.

Sprites made by thegreatjorge.

### CODE REVIEW

File: code/game/objects/items/weapons/landmines.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		icon_state = "[icon_state]"
+	else
+		icon_state = "[icon_state]_on"
+
+/obj/item/weapon/landmine/verb/hide_under()
+	set src in oview(1)
+	set name = "Hide"
+	set category = "Object"
+
+	if(usr.stat)
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: This should probably be `use_check()`.

File: code/game/objects/items/weapons/landmines.dm [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		return
+
+	layer = TURF_LAYER+0.2
+	usr << "<span class='notice'>You hide \the [src].</span>"
+
+/obj/item/weapon/landmine/proc/can_use(mob/user)
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: Why not just use `use_check()`?

File: code/game/objects/items/weapons/landmines.dm [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	if(use_check(user, USE_DISALLOW_SILICONS))
+		return
+
+	layer = TURF_LAYER+0.2
+	usr << "<span class='notice'>You hide \the [src].</span>"
+
+
+/obj/item/weapon/landmine/attack_self(mob/user as mob)
+	..()
+	if(!deployed && use_check(user, USE_DISALLOW_SILICONS))
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: This is probably the opposite you want, `use_check()` returns `FALSE` on success.

**Comment by Sindorman** [comment]
Landmines are usually usually triggered when you step on them, aka pressure sensor. These appears to use proximity. With regular mines you can disarm them, but these appears to be not being able to be disarmed.
FYI: if you will make them craftable somehow please then add proximity sensor to it, so it makes sense.

**Comment by Lohikar** [comment]
Crossed is not proximity, these trigger when your mob enters the same tile as the mine.

**Comment by Lohikar** [comment]
Landmines can't be disarmed IRL.

**Comment by Sindorman** [comment]
Exactly, there is a bombsquad kit. Plus what about if you drop a heavy object on same tile, then it shouldn't blow up as well.

**Comment by Alberyk** [comment]
They can be disarmed via shooting one or explosions, if not by forcing something big enough to walk over one.

File: code/game/objects/items/weapons/landmines.dm [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	spark(src, 3, alldirs)
+	for (var/turf/simulated/floor/target in range(1,src))
+		if(!target.blocks_air)
+			target.assume_gas("phoron", 30)
+
+			target.hotspot_expose(1000, CELL_VOLUME)
+
+	qdel(src)
+
+/obj/item/weapon/landmine/n2o
```
Comment (Arrow768) [inline code/game/objects/items/weapons/landmines.dm]: I think it would be nice to add the n2o and the emp to the uplink aswell.

Comment (Alberyk) [reply code/game/objects/items/weapons/landmines.dm]: Added them to the uplink as well.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 5]
```diff
@@ -49,6 +49,21 @@
...
+	item_cost = 5
+	path = /obj/item/weapon/landmine
+
+/datum/uplink_item/item/visible_weapons/emplandmine
+	name = "EMP Land Mine"
+	item_cost = 4
+	path = /obj/item/weapon/landmine/emp
+
+/datum/uplink_item/item/visible_weapons/sleepylandmine
+	name = "N2o Land Mine"
```
Comment (Lohikar) [inline code/datums/uplink/highly visible and dangerous weapons.dm]: This should be stylized as `N2O`, not `N2o`.

File: code/game/objects/items/weapons/landmines.dm [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+				)
+
+			deployed = 1
+			user.drop_from_inventory(src)
+			update_icon()
+			anchored = 1
+
+/obj/item/weapon/landmine/proc/trigger(mob/living/L)
+	spark(src, 3, alldirs)
+	if(L)
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: This if is redundant, `ishuman()` will return FALSE if `L` is `null`.

File: code/game/objects/items/weapons/landmines.dm [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+		if(ishuman(L))
+			L.Weaken(2)
+	explosion(loc, 0, 1, 2, 3)
+	qdel(src)
+
+/obj/item/weapon/landmine/Crossed(AM as mob|obj)
+	if(deployed && isliving(AM))
+		var/mob/living/L = AM
+		if(L.mob_size >= 5)
+			L.visible_message(
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: Maybe add a click sound for blind people (3rd param)?

File: code/game/objects/items/weapons/landmines.dm [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+				)
+			trigger(L)
+	..()
+
+/obj/item/weapon/landmine/attack_hand(mob/user as mob)
+	if(deployed && !use_check(user, USE_DISALLOW_SILICONS))
+		user.visible_message(
+				"<span class='danger'>[user] triggers \the [src].</span>",
+				"<span class='danger'>You trigger \the [src]!</span>"
+				)
```
Comment (Lohikar) [inline code/game/objects/items/weapons/landmines.dm]: 3rd param here too maybe.

File: code/game/objects/items/weapons/landmines.dm (hunk 7 above)
Comment (Alberyk) [reply code/game/objects/items/weapons/landmines.dm]: Fixed.

File: code/datums/uplink/highly visible and dangerous weapons.dm (hunk 5 above)
Comment (Alberyk) [reply code/datums/uplink/highly visible and dangerous weapons.dm]: Fixed.

File: code/game/objects/items/weapons/landmines.dm (hunk 4 above)
Comment (Alberyk) [reply code/game/objects/items/weapons/landmines.dm]: Fixed.

**Pull request merged**